jmclient/jmclient/jsonrpc.py

```diff
@@ -54,24 +56,24 @@ class JsonRpc(object):
   to connect to Bitcoin.
   """
 
-    def __init__(self, host, port, user, password, url=""):
+    def __init__(self, host: str, port: int, user: str, password: str, url: str = "") -> None:
         self.host = host
         self.port = int(port)
         self.conn = http.client.HTTPConnection(self.host, self.port)
         self.authstr = "%s:%s" % (user, password)
         self.url = url
         self.queryId = 1
 
-    def setURL(self, url):
+    def setURL(self, url: str) -> None:
         self.url = url
 
-    def queryHTTP(self, obj):
+    def queryHTTP(self, obj: object) -> object:
```

Comment:
Can't we replace both `object` with `dict` here? Return should be `Union[dict, str]`? Do we expect this to return anything other than a dict or a string?

Reply:
Agreed, see no ambiguity here. Return must be `dict` or `str` (i.e. it specifically can't be `None` since the only way to break out of the infinite loop is if an exception is thrown, and `Exception` is caught at the end.

Reply:
Agree about return type, but, by looking at JSON-RPC specs, `obj` argument should be `Union[dict, list]`, no?

Reply:
Ahh, no, it's definitely `dict`, as it is full request body, not just `params`, here.

Reply:
Addressed in 15c099a51e54f6e8666e07affceb208942ceb2ba